src/overlays/actors/ovl_En_Kitan/z_en_kitan.c

```diff
@@ -27,50 +30,364 @@ ActorInit En_Kitan_InitVars = {
+}
+
+s32 EnKitan_CanTalk(EnKitan* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    if (player->stateFlags1 & PLAYER_STATE1_800000) {
+        return false;
+    }
+    if (Player_IsFacingActor(&this->actor, 0x3000, play) && Actor_IsFacingPlayer(&this->actor, 0x3000) &&
+        this->actor.xzDistToPlayer < 120.0f) {
+        return true;
+    }
+    return false;
+}
+
+u16 EnKitan_GetQuestionMessageId(EnKitan* this) {
+    s32 i = 0;
+
+    //! @bug This loop was likely meant to have a max iteration cap of 1000, but the condition is messed up so it will
+    //! loop forever until a question that was not yet asked is found. This is fine as the number of possible questions
+    //! is larger than the number of questions the player is required to answer, assuming the random number generator
+    //! can output the full range of values which is the case.
+    while (true) {
+        s32 rand = Rand_ZeroFloat(30.0f);
+
+        // Keep track of which questions have already been asked with a bitset
+        if (!(this->textBitSet & (1 << rand))) {
+            this->textBitSet |= (1 << rand);
+            // 0x04B6 is the start of the question + answer choice textboxes, each question textbox is followed by the
+            // choice textbox containing the answer choices
+            return 0x04B6 + 2 * rand;
+        }
+
+        i++;
+        if (i > 1000) {
+            continue;
+        }
+    }
+}
+
+void EnKitan_Leave(EnKitan* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+
+    if (this->timer > 0) {
+        // Scale down
+        this->timer--;
+        this->actor.scale.x *= 0.7f;
+        Actor_SetScale(&this->actor, this->actor.scale.x);
+    } else {
+        Actor_Kill(&this->actor);
+    }
+}
+
+void EnKitan_TalkAfterGivingPrize(EnKitan* this, PlayState* play) {
+    s32 pad;
+
+    SkelAnime_Update(&this->skelAnime);
+
+    if (Actor_TalkOfferAccepted(&this->actor, &play->state)) {
+        this->actionFunc = EnKitan_Talk;
+        Message_ContinueTextbox(play, 0x04B5);
+        this->actor.flags &= ~ACTOR_FLAG_10000;
+        Animation_MorphToLoop(&this->skelAnime, &gKeatonChuckleAnim, -5.0f);
+    } else {
+        Actor_OfferTalkExchange(&this->actor, play, 1000.0f, 1000.0f, PLAYER_IA_MINUS1);
+    }
+}
+
+void EnKitan_WaitForPrizeTextboxClosed(EnKitan* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+
+    if (Actor_TextboxIsClosing(&this->actor, play)) {
+        this->actor.flags |= ACTOR_FLAG_10000;
+        this->actionFunc = EnKitan_TalkAfterGivingPrize;
+        Actor_OfferTalkExchange(&this->actor, play, 1000.0f, 1000.0f, PLAYER_IA_MINUS1);
+    }
+}
+
+void EnKitan_OfferPrize(EnKitan* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+
+    if (Actor_HasParent(&this->actor, play)) {
+        this->actor.parent = NULL;
+        this->actionFunc = EnKitan_WaitForPrizeTextboxClosed;
+        gSaveContext.save.saveInfo.weekEventReg[0x4F] |= 0x80;
```

Comment:
```suggestion
        SET_WEEKEVENTREG(WEEKEVENTREG_79_80);
```

You can do `./tools/weekeventregconvert.py src/overlays/actors/ovl_En_Kitan/z_en_kitan.c` to get all these weekeventreg things converted automatically.